signals/parser/schema.py

```diff
@@ -28,6 +29,25 @@ def create_objects(self, object_json):
             data_object = DataObject(object_name, data_object_json[object_name])
             self.data_objects[data_object.name] = data_object
 
+    # Note: Why do we call them "apis" in method names, but then call them "urls" in the schema/code
+    def validate_apis_and_objects(self):
+        for url in self.urls:
+            for endpoint in URL.URL_ENDPOINTS.keys():
+                api = getattr(url, endpoint, None)
+                if api:
+                    if getattr(api, 'parameters_object', None):
```

Comment:
We should reduce all of these `if`'s to a single call that calls another method and passes in the object attribute we're looking for. Even better we have a list of them `['parameters_object', 'request_object', 'response_object']` and we loop through calling that method on each of them.